vcd/resource_vcd_solution_add_on_test.go

```diff
@@ -0,0 +1,229 @@
+//go:build slz || api || functional || ALL
+
+package vcd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func TestAccSolutionAddon(t *testing.T) {
+	preTestChecks(t)
+	skipIfNotSysAdmin(t)
+
+	if checkVersion(testConfig.Provider.ApiVersion, "< 37.1") {
+		t.Skipf("Solution Landing Zones are supported in VCD 10.4.1+. Skipping")
+	}
+
+	if testConfig.VCD.Catalog.NsxtCatalogAddonDse == "" {
+		t.Skipf("Add-On config value not specified ")
+	}
+
+	vcdClient := createTemporaryVCDConnection(true)
+	org, err := vcdClient.GetOrgByName(testConfig.VCD.Org)
+	if err != nil {
+		t.Fatalf("error creating temporary VCD connection: %s", err)
+	}
+
+	catalog, err := org.GetCatalogByName(testConfig.VCD.Catalog.NsxtBackedCatalogName, false)
+	if err != nil {
+		t.Fatalf("error retrieving catalog: %s", err)
+	}
+
+	localAddOnPath, err := fetchCacheFile(catalog, testConfig.VCD.Catalog.NsxtCatalogAddonDse, t)
+	if err != nil {
+		t.Fatalf("error finding Solution Add-On cache file: %s", err)
+	}
+
+	params := StringMap{
+		"Org":     testConfig.VCD.Org,
+		"VdcName": testConfig.Nsxt.Vdc,
+
+		"TestName":            t.Name(),
+		"CatalogName":         testConfig.VCD.Catalog.NsxtBackedCatalogName,
+		"RoutedNetworkName":   testConfig.Nsxt.RoutedNetwork,
+		"IsolatedNetworkName": testConfig.Nsxt.IsolatedNetwork,
+
+		"AddonIsoPath": localAddOnPath,
+	}
+	testParamsNotEmpty(t, params)
+
+	params["FuncName"] = t.Name() + "step1"
+	configText1 := templateFill(testAccSolutionAddonStep1, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 1: %s", configText1)
+
+	params["FuncName"] = t.Name() + "step2"
+	configText2 := templateFill(testAccSolutionAddonStep2, params)
```

Comment:
When running as binary test, step 2 does little more than step 1. If used in isolation, it is completely equivalent. We should skip one of them in binary tests

Reply:
Skipping it